packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/funding-source/index.hbs

```diff
@@ -1,19 +1,102 @@
+              @icon="avatar-default"
+              @title="Account 1"
+              @address={{this.layer2Network.walletInfo.firstAddress}}
+              data-test-funding-source-account
+            />
+          </:outer>
+          <:inner>
+            <CardPay::AccountDisplay
+              @isComplete={{@isComplete}}
+              @inner={{true}}
+              @icon="depot"
+              @title="DEPOT:"
+              @address={{this.fetchDepotTask.last.value.address}}
+              data-test-funding-source-depot-inner
+            />
+          </:inner>
+        </CardPay::NestedItems>
+      </CardPay::LabeledValue>
+    </ActionCardContainer::Section>
+    <ActionCardContainer::Section>
+      <CardPay::LabeledValue @label="BALANCE:" class="funding-source__field">
+        <CardPay::NestedItems
+          class="funding-source__field-info"
+          @noBorder={{not @isComplete}}
+          @noIndent={{not @isComplete}}
+        >
+          <:outer>
+            <CardPay::AccountDisplay
+              @isComplete={{@isComplete}}
+              @icon="depot"
+              @title="DEPOT:"
+              @address={{this.fetchDepotTask.last.value.address}}
+              data-test-funding-source-depot-outer
+            />
+          </:outer>
+          <:inner>
+            {{#if @isComplete}}
+              <CardPay::AccountDisplay
+                @isComplete={{@isComplete}}
+                @icon={{this.selectedToken.icon}}
+                @balance={{this.selectedToken.balance}}
+                @symbol={{this.selectedToken.symbol}}
+                data-test-funding-source-token
+              />
+            {{else}}
+              <div class="funding-source__nested-dropdown">
+                <PowerSelect
+                  class="funding-source__dropdown"
+                  @options={{this.tokens}}
+                  @selected={{this.selectedToken}}
+                  @onChange={{this.chooseSource}}
+                  @renderInPlace={{true}}
+                  @verticalPosition="below"
+                  data-test-funding-source-dropdown={{this.selectedToken.symbol}}
+                as |token|>
+                  <CardPay::AccountDisplay
+                    class={{cn "funding-source__dropdown-option" (if (eq token.symbol this.selectedToken.symbol) "funding-source__dropdown-option--selected")}}
+                    @isComplete={{@isComplete}}
+                    @icon={{token.icon}}
+                    @balance={{token.balance}}
+                    @symbol={{token.symbol}}
+                    data-test-funding-source-token-option={{token.symbol}}
+                  />
+                </PowerSelect>
+              </div>
+            {{/if}}
+          </:inner>
+        </CardPay::NestedItems>
+      </CardPay::LabeledValue>
+    </ActionCardContainer::Section>
+  </ActionCardContainer::Section>
   <Boxel::ActionChin
     @state={{if @isComplete "memorialized" "default"}}
     @disabled={{@frozen}}
```

Comment:
```suggestion
    @disabled={{or @frozen (not this.selectedToken)}}
```

Reply:
Otherwise, `not this.selectedToken` can override `@frozen` after completion/cancelation of the workflow

Reply:
Alternatively, apply `@frozen` to the Edit button?

Reply:
Similar case in https://github.com/cardstack/cardstack/commit/a1da448ece3bb536b49a1155be79568c331497e4

Reply:
Removed the other disabled condition since there's already a preselected token. Haven't handled the edge cases though